### DISCUSSION TITLE
Bug in c header file

### DISCUSSION THREAD

**Discussion by chrisbohens** [esynr3z/corsair#133] [Questions]
Hi. There is a bug in the c header file generation with the reserved-padding. It occurs when only the last bit of a field is not used (reserved) and the one before is. This results in a c-struct with 31 bits instead of 32, which is not good.

I can't test it right now, but my first guess is that it is a off-by-one mistake in the c_header.j2 file. The 

```j2
{%if tmp.lsb < config.data_width - 1%}
```

should probably be 

```j2
{%if tmp.lsb < config.data_width%}
```

instead. Maybe someone can confirm?

In the generated md-file the padded reserved bit is generated correctly.

**Comment by esynr3z** [comment]
Hi, thank you for the report. Need to investigate.

Interesting that you refer the line that was changed within previous wave of fixes:
- #25 
- #26
